### PR TITLE
Keycloak auth server

### PR DESCRIPTION
**Pull request by pdevito3**
Replaces the duende auth server with keycloak and pulumi

### CODE REVIEW

**Comment by ogkushmin** [comment]
Is there a way to still gen up duende instead of keycloak?

**Comment by pdevito3** [comment]
not at the moment. the big motivators being increased maintenance and lack of a robust duende setup that was actually going to be useful for anything more than demo flow, which the keycloak setup does alongside having a more robust option ootb.

i started working on [a more robust project](https://github.com/pdevito3/identity-server-sandbox) that would turn to scaffolding but it was becoming a huge lift. I do have [a pending request in with duende](https://github.com/DuendeSoftware/Samples/issues/75) on a more plug and play option that they seemed receptive to and could be integrated in the future.

bottom line is that it was just becoming too big of an effort to get something of quality in at this point. i would love to get something more robust in at some point though.

at the moment, if you want to use it, you might just want to downgrade craftsman and do an `add authserver` to get it in or grab a template project and update the auth config.
